File: Database.Setlists/DataObjects/SetlistAct.cs
namespace Database.Setlists.DataObjects;

public class SetlistAct
{
    /// <summary>
    /// ID of the setlist
    /// </summary>
    public uint SetlistId { get; set; }
    
    
    /// <summary>
    /// Unique ID of the setlist
    /// </summary>
    public uint ActNumber { get; set; }


    /// <summary>
    /// Title of this act
    /// </summary>
    public string? Title { get; set; }
}

File: frontend/lp-calendar-web/src/app/data/setlists/setlist-act.ts
import {SetlistActDto} from '../../modules/lpshows-api';
import {SetlistEntry} from './setlist-entry';

export class SetlistAct {
  setlistId!: number;
  actNumber!: number;
  title!: string;


  static fromDto(dto: SetlistActDto): SetlistAct {
    return {
      setlistId: dto.setlistId!,
      actNumber: dto.actNumber!,
      title: dto.title ?? `Act ${dto.actNumber}`,
    };
  }
}

export class SetlistActWithEntries extends SetlistAct {
  entries: SetlistEntry[] = [];

  static withEntries(act: SetlistAct, entries: SetlistEntry[]): SetlistActWithEntries {
    return {
      setlistId: act.setlistId!,
      actNumber: act.actNumber!,
      title: act.title ?? `Act ${act.actNumber}`,
      entries: entries
    };
  }
}
